endpoints/openrtb2/amp_auction.go

```diff
@@ -276,6 +278,94 @@ func (deps *endpointDeps) loadRequestJSONForAmp(httpRequest *http.Request) (req
 		*req.Imp[0].Secure = 1
 	}
 
+	// Override query params
+	if width, err := strconv.ParseUint(httpRequest.FormValue("w"), 10, 64); err == nil {
+		if req.Imp[0].Banner != nil {
+
+			*req.Imp[0].Banner.W = width
+		}
+	}
+
+	if height, err := strconv.ParseUint(httpRequest.FormValue("h"), 10, 64); err == nil {
+		if req.Imp[0].Banner != nil {
+			*req.Imp[0].Banner.H = height
+		}
+	}
+
+	if overrideWidth, err := strconv.ParseUint(httpRequest.FormValue("ow"), 10, 64); err == nil {
+		// override w
+		if req.Imp[0].Banner != nil {
+			*req.Imp[0].Banner.W = overrideWidth
+		}
+	}
+
+	if overrideHeight, err := strconv.ParseUint(httpRequest.FormValue("oh"), 10, 64); err == nil {
+		// override h
+		if req.Imp[0].Banner != nil {
+			*req.Imp[0].Banner.H = overrideHeight
+		}
+	}
+
+	multiSize := httpRequest.FormValue("ms")
+	if multiSize != "" {
+		// override w and h with format
+		req.Imp[0].Banner.H = nil
+		req.Imp[0].Banner.W = nil
```

Comment:
I'm not sure this is quite right... at least, I'm finding some things which suggests that AMP treats these as "extra" sizes. For example: https://github.com/ampproject/amphtml/blob/master/examples/doubleclick-multisize.amp.html#L45

In OpenRTB, you're allowed to set `w`, `h`, _and_ `format`... so it might be better to leave these set?